core/domain/state_domain.py

```diff
@@ -1697,7 +1666,7 @@ def update_interaction_answer_groups(self, answer_groups_list):
                     else:
                         try:
                             normalized_param = param_type.normalize(value)
-                        except TypeError:
```

Comment:
Why this change?

Reply:
I don't think normalize raises a TypeError. Also it may produce other types of exception which is not caught by TypeError

Reply:
I assume we must have guarded the method with other exceptions. 
>  it may produce other types of exception
Can you please give a few examples.

Reply:
For example, I added this test `test_cannot_update_answer_groups_with_invalid_rule_input_value` which does not work for TypeError but works for general "Exception"